### PR TITLE
Eliminate nullness errors in KafkaIO

### PR DESCRIPTION
**Pull request by kennknowles**
**Please** add a meaningful description for your change here

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [x] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [x] Mention the appropriate issue in your description (for example: "addresses #123), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment "fixes #<ISSUE NUMBER>" instead.
 - [x] Update `CHANGES.md` with noteworthy changes.
 - [x] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/#make-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

**Comment by kennknowles** [comment]
R: @johnjcasey 

We discussed some cases.

This one I did a lot more "just add an assertion" style fixes, whereas usually I like to refactor things to eliminate the possibility of any error occurring. If you can spot places where it simply doesn't make sense to have a nullable field, please comment.

File: sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/ConsumerSpEL.java [hunk 1]
```diff
@@ -117,7 +118,7 @@ public static void evaluateSeek2End(Consumer<?, ?> consumer, TopicPartition topi
   }
 
   public static void evaluateAssign(
-      Consumer<?, ?> consumer, Collection<TopicPartition> topicPartitions) {
+      @Nullable Consumer<?, ?> consumer, @Nullable Collection<TopicPartition> topicPartitions) {
```
[the remark this answers is not in the document]
Comment (kennknowles) [reply sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/ConsumerSpEL.java]: Done

File: sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/KafkaExactlyOnceSink.java [hunk 2]
```diff
@@ -691,33 +700,36 @@ void insert(int shard, ShardWriter<K, V> writer) {
...
-                spec.getSinkGroupId(),
-                ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG,
-                ByteArrayDeserializer.class,
-                ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG,
-                ByteArrayDeserializer.class));
+    SerializableFunction<Map<String, Object>, ? extends Consumer<?, ?>> consumerFactoryFn =
+        Preconditions.checkArgumentNotNull(spec.getConsumerFactoryFn());
+
+    Map<String, Object> consumerConfig = new HashMap<>();
+    Object bootStrapServersConfig =
```
[the remark this answers is not in the document]
Comment (kennknowles) [reply sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/KafkaExactlyOnceSink.java]: Done

**Comment by kennknowles** [comment]
And in getting tests to pass, I found that they simply didn't call `setup()`, so now the tests are more accurate to real usage.

**Comment by johnjcasey** [comment]
> And in getting tests to pass, I found that they simply didn't call `setup()`, so now the tests are more accurate to real usage.

That is very appreciated

**Comment by kennknowles** [comment]
run java precommit

**Comment by kennknowles** [comment]
run java precommit

**Comment by kennknowles** [comment]
Run Python_PVR_Flink PreCommit

**Comment by kennknowles** [comment]
Run Python_PVR_Flink PreCommit

**Comment by kennknowles** [comment]
seeing "could not find valid docker environment" across a lot of PRs, hmm

**Comment by kennknowles** [comment]
Run Java PreCommit

**Comment by kennknowles** [comment]
Run Java PreCommit

**Comment by kennknowles** [comment]
Run Java PreCommit

**Comment by kennknowles** [comment]
Run Python_PVR_Flink PreCommit

**Comment by kennknowles** [comment]
no error in the logs, and scan failed to upload :-/

**Comment by kennknowles** [comment]
Run Python_PVR_Flink PreCommit

**Comment by kennknowles** [comment]
Please take another look now that tests are green.

Reminder of policy: when a committer is the author, they are trusted to choose the best reviewer, including not-yet-committers. I think you are the best choice for this region of the code. And thanks for the review so far!

**Pull request merged**